ssp

```diff
@@ -7972,18 +7975,32 @@ sub _http_get {
         MaxReply            => 50_000,
         MultiHomed          => 1,
         Path                => '/',
-        Port                => 80,
         Proto               => 'tcp',
         ReportTimeout       => 0,
         ReportTimeoutHeader => '',
-        Timeout             => 2,
+        SSL                 => 0,
+        SSLLoadFailWarning  => undef,
+        SSLVerifyHost       => 1,
         Tries               => 1,
         WantHeaders         => 0,
         %opts_in
     );
     my $try = 0;
     my $reply;
     die unless defined $opts{Host};
+    if ( $opts{SSL} ) {
+        return unless load_module_with_fallbacks(
+            'needed_subs'  => [qw(start_SSL import)],
+            'modules'      => [qw(IO::Socket::SSL)],
+            'fail_warning' => $opts{SSLLoadFailWarning},
+        );
+        $opts{Port}    = 443 if not defined $opts{Port};
+        $opts{Timeout} = 3   if not defined $opts{Timeout};
```

Comment:
Why do we only allow 3 seconds here as opposed to 5 etc?  I assume this is just to lock down runtime.

Reply:
Yes, mainly to keep runtime down, though this is actually 1 second longer than non-SSL to allow for the additional handshake overhead.